apps/ui/lib/components/Event/Message/DisplayReactions.tsx

```diff
@@ -0,0 +1,165 @@
+
+	const reactions = message.links?.['has attached element']?.filter(
+		(contract) => {
+			return contract.type === 'reaction@1.0.0';
+		},
+	);
+
+	const [groupReactions, setGroupReactions] = useState(
+		{} as { [key: string]: Contract[] },
+	);
+
+	const { sdk } = useSetup()!;
+	// TODO: Remove this workaround when this is resolved
+	// https://github.com/product-os/autumndb/issues/1333
+	React.useEffect(() => {
+		if (!reactions) return;
+		sdk
+			.query({
+				type: 'object',
+				$$links: {
+					'was created by': {
+						type: 'object',
+					},
+				},
+				properties: {
+					id: {
+						enum: reactions.map((reaction) => reaction.id),
+					},
+				},
+			})
+			.then((reactions) => {
+				const result = reactions.reduce((map, reaction) => {
+					const symbol = reaction.data.reaction as string;
+					map[symbol] = map[symbol] || [];
+					map[symbol].push(reaction.links?.['was created by']?.[0]!);
+					return map;
+				}, {} as { [key: string]: Contract[] });
+				setGroupReactions(result);
+			});
+	}, [reactions]);
+
+
+
+	// TODO: Enable this when this is resolved
+	// https://github.com/product-os/autumndb/issues/1333
+
+	// const groupedReactions = reactions.reduce((map, reaction) => {
+	// 	console.log(map, reaction);
+	// 	const symbol = reaction.data.reaction as string;
+	// 	map[symbol] = map[symbol] || [];
+	// 	map[symbol].push(reaction.links?.['was created by']?.[0]!);
+	// 	return map;
+	// }, {} as { [key: string]: Contract[] });
+
+	const handleClick = React.useCallback(
+		async (symbol) => {
+			const [reaction] = await sdk.query({
+				type: 'object',
+				required: ['type', 'data'],
+				properties: {
+					type: {
+						const: 'reaction@1.0.0',
+					},
+					data: {
+						type: 'object',
+						required: ['reaction'],
+						properties: {
+							reaction: {
+								const: symbol,
+							},
+						},
+					},
+				},
+				$$links: {
+					'was created by': {
+						type: 'object',
+						required: ['id'],
+						properties: {
+							id: {
+								const: currentUser.id,
+							},
+						},
+					},
+					'is attached to': {
+						type: 'object',
+						required: ['id'],
+						properties: {
+							id: {
+								const: message.id,
+							},
+						},
+					},
+				},
+			});
+
+			if (reaction) {
+				await sdk.card.remove(reaction.id, reaction.type); // check for the parameter order
+			} else {
+				const newReaction = await sdk.card.create({
+					type: 'reaction',
+					data: {
+						reaction: symbol,
+					},
+				});
+
+				await sdk.card.link(message, newReaction, 'has attached element');
+			}
+		},
+		[reactions, message],
+	);
+
+		if (!reactions || !reactions.length) {
+			return null;
+		}
+
+	return (
+		<Wrapper>
+			{Object.entries(groupReactions).map(([symbol, users]) => {
+				return (
+					<Button onClick={() => handleClick(symbol)} p={1} mx={0.5}>
+						<Emoji
+							symbol={symbol}
+							count={users.length}
+							tooltip={{
```

Comment:
The tooltip should be placed on the `Button` element, so that you get the tooltip on hover of the control element, rather than just the inner `Emoji` component